### PR TITLE
feat: Adds the ability to add, delete and customize questionnaire fields

### PR DESCRIPTION
**Pull request by JeremyRudman**
Made the basics for adding hackathon specific questions. Still need to add more options for the questions, improve the UI and fix some bugs

chrome_2020-08-25_13-37-56
chrome_2020-08-25_13-15-37
chrome_2020-08-25_13-15-21
chrome_2020-08-25_13-15-07
chrome_2020-08-24_12-22-32

### CODE REVIEW

**Comment by cbaudouinjr** [comment]
Loving this so far, thought of some additional things we should add/watch out for. Right now a questionnaire is normally split in two, the identity portion and the event-specific questions portion. What if we seeded the already existing event-specific questions but now users can remove them/add in others? This would be great for events that want to go virtual but don't need questions such as travel, dietary restrictions, etc.

**Comment by cbaudouinjr** [comment]
Known issues:

- [ ] Required field not being enforced

**Comment by JeremyRudman** [comment]
This to do:
- [ ] Make proper extra question index page
- [ ] Add more options extra questions 
- [ ] Get date and time data types to work

**Comment by JeremyRudman** [comment]
image

**Comment by JeremyRudman** [comment]
made the index page not horrible and added placeholder option to questions
image

File: app/models/extra_question.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
+class ExtraQuestion < ApplicationRecord
+  validates_presence_of :question, :data_type
+
```
Comment (cbaudouinjr) [inline app/models/extra_question.rb:3]: What is the association?

**Review by cbaudouinjr — CHANGES_REQUESTED** [review]

File: app/models/extra_question.rb (hunk 1 above)
Comment (JeremyRudman) [reply app/models/extra_question.rb:3]: I am not sure rails associations would work since the ID is stored as part of a serialized hash in questionnaire

**Comment by peterkos** [comment]
Given we're planning to completely refactor the frontend in [3.0](https://github.com/codeRIT/hackathon-manager/issues?q=is%3Aopen+is%3Aissue+label%3A3.0), this is a lot of code built around a solution (SimpleForm) that we're going to remove in a few months. There are still a lot of UX issues with regards to the questionnaire, so I feel that it'd be best if we don't merge this in, and not risk further bugs+time maintaining SimpleForm-related issues.

Let's keep this PR open at least so that in case we decide we need it, we can merge in as `2.3` or something. Maybe have a parallel 2.3 release while we work on 3.0?

**Comment by peterkos** [comment]
Closing in favor of 3.0 work; written down in Notion to go back to!

**Pull request closed without merging**